hazelcast/src/test/java/com/hazelcast/jet/core/JobTimeoutTest.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.core;
+
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.jet.Job;
+import com.hazelcast.jet.config.JobConfig;
+import com.hazelcast.test.annotation.ParallelJVMTest;
+import com.hazelcast.test.annotation.QuickTest;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+import java.util.concurrent.CancellationException;
+
+import static org.junit.Assert.assertEquals;
+
+@Category({QuickTest.class, ParallelJVMTest.class})
+public class JobTimeoutTest extends JetTestSupport {
+
+    @Test
+    public void when_lightJobIsCompletedAfterTimeout_jobIsCancelled() {
+        final HazelcastInstance hz = createHazelcastInstance();
+        final DAG dag = new DAG();
+        dag.newVertex("stuck", StuckSource::new);
```

Comment:
```suggestion
        dag.newVertex("stuck", () -> new MockP().streaming());
```
Again on line 50 & more.